test/integration_test/applicationbackup_test.go

```diff
@@ -47,114 +61,118 @@ func triggerBackupRestoreTest(
+			require.NoError(t, err, "Error waiting for restore to complete.")
 
-		err = schedulerDriver.WaitForRunning(restoreCtx, defaultWaitTimeout, defaultWaitInterval)
-		require.NoError(t, err, "Error waiting for restore to complete.")
+			logrus.Infof("Restore completed.")
 
-		logrus.Infof("Restore completed.")
+			// Validate that restore results in restoration of correct apps based on whether all apps were expected or not
+			err = schedulerDriver.WaitForRunning(preBackupCtx, defaultWaitTimeout, defaultWaitInterval)
+			require.NoError(t, err, "Error waiting for restore to complete.")
 
-		// Validate that restore results in restoration of correct apps based on whether all apps were expected or not
-		err = schedulerDriver.WaitForRunning(preBackupCtx, defaultWaitTimeout, defaultWaitInterval)
-		require.NoError(t, err, "Error waiting for restore to complete.")
+			logrus.Infof("App validations after restore completed.")
 
-		logrus.Infof("App validations after restore completed.")
+		} else { // Since backup is expected to fail, reducing the wait time here to catch the error faster
+			err = schedulerDriver.AddTasks(ctxs[0], scheduler.ScheduleOptions{AppKeys: appBackupKey})
+			require.NoError(t, err, "Error backing-up apps")
+			err = schedulerDriver.WaitForRunning(ctxs[0], defaultWaitTimeout/5, defaultWaitInterval)
+			require.Error(t, err, "Backup expected to fail in test: %s.", t.Name())
+		}
 
-	} else { // Since backup is expected to fail, reducing the wait time here to catch the error faster
-		err = schedulerDriver.AddTasks(ctxs[0], scheduler.ScheduleOptions{AppKeys: appBackupKey})
-		require.NoError(t, err, "Error backing-up apps")
-		err = schedulerDriver.WaitForRunning(ctxs[0], defaultWaitTimeout/5, defaultWaitInterval)
-		require.Error(t, err, "Backup expected to fail in test: %s.", t.Name())
-	}
+		ctxs = append(ctxs, restoreCtx)
+		if (backupAllAppsExpected && backupSuccessExpected) || !backupSuccessExpected {
+			destroyAndWait(t, ctxs)
+		} else if !backupAllAppsExpected && backupSuccessExpected {
+			// Some apps might have been already destroyed, destroy the remaining
+			destroyAndWait(t, []*scheduler.Context{postRestoreCtx})
 
-	ctxs = append(ctxs, restoreCtx)
-	if (backupAllAppsExpected && backupSuccessExpected) || !backupSuccessExpected {
-		destroyAndWait(t, ctxs)
-	} else if !backupAllAppsExpected && backupSuccessExpected {
-		// Some apps might have been already destroyed, destroy the remaining
-		destroyAndWait(t, []*scheduler.Context{postRestoreCtx})
+		}
+		err = k8s.Instance().DeleteBackupLocation(currBackupLocation.Name, currBackupLocation.Namespace)
+		require.NoError(t, err, "Failed to delete backuplocation: %s for location %s.", currBackupLocation.Name, string(location), err)
 	}
 }
 
 func createBackupLocation(
 	t *testing.T,
 	name string,
 	namespace string,
-	locationtype storkv1.BackupLocationType,
-	configMapName string,
+	locationtype string,
```

Comment:
Keep the type as `storkv1.BackupLocationType`, `getBackupLocationType` can also take in that type instead of string

Reply:
Done.